EIPS/eip-network_upgrade_windows.md

```diff
@@ -0,0 +1,171 @@
+Dao Fork, Tangerine Whistle, and Spurious Dragon.
+
+This EIP provides neither guidance nor restrictions to the development and
+deployment of these emergency hard forks. These upgrades are typically launched
+promptly after a solution to the systemic risk is agreed upon between the client
+implementors.
+
+### Network Upgrade Naming
+
+The choice of a name for a network upgrade can help communicate what type of
+network upgrade it is. There have been patterns used in the past that if
+continued would help communicate this.
+
+Roadmap Network Upgrades have historically been given single word names. The
+first three network upgrades were Frontier, Homestead, and Metropolis.
+Metropolis was split into two upgrades, Byzantium and Constantinople. As of the
+writing of this EIP the next network upgrade will be called Istanbul.
+
+Critical Network Upgrades have historically been two word phrases. Sometimes the
+name applies to the issue at hand, sometimes it do not. Previous Critical
+Network upgrade names have been Dao Fork, Tangerine Whistle, Spurious Monkey,
+and Constantinople Fix.
+
+### Network Upgrade Block Number Choice
+
+When choosing an activation block the number can be used to communicate the role
+of a particular network in the Ethereum Ecosystem.
+
+To date all Mainnet activation blocks have ended in three or more zeros,
+including Critical Network Upgrades. Ropsten and Kovan initially started with
+three zeros but switched to palindromic numbers. Rinkeby has always had
+palindromic activation blocks. Goerli has yet to perform a network upgrade.
+
+To continue this pattern network upgrade activation block numbers for mainnet
+deployment should chose a number whose base 10 representation ends with three or
+more zeros.
+
+For testnet deployments network operators are encouraged to choose a block
+activation number that is a palindrome in base 10.
+
+## Rationale
+
+The rationale for defining launch windows is to give business running Ethereum
+infrastructure a predictable schedule for when upgrades may or may not occur.
+Knowing when a upgrade is not going to occur gives the businesses a clear time
+frame within which to perform internal upgrades free from external changes. It
+also provides a timetable for developers and IT professionals to schedule time
+off against.
+
+## Backwards Compatibility
+
+Except for the specific launch windows the previous network upgrades would have
+complied with these policies. Homestead, Byzantium, and Constantinople would
+have been Roadmap Network Upgrades. There were no Priority Network Upgrades,
+although Spurious Dragon would have been a good candidate. Dao Fork was a
+Critical Network Upgrade in response to TheDao. Tangerine Whistle and Spurious
+Dragon were critical upgrades in response to the Shanghai Spam Attacks.
+Constantinople Fix (as it is termed in the reference tests) was in response to
+the EIP-1283 security issues.
```

Comment:
An analysis of the dates when previous upgrades were proposed/performed may be helpful to the dates discussion above.